e2e/analysis/test_case.go

```diff
@@ -0,0 +1,673 @@
+	Command        Command
+	Expected       Expected
+}
+
+func (tc TestCase) RunAnalysisTestCase() (*gexec.Session, error) {
+	isWindows := runtime.GOOS == "windows"
+	if isWindows || !tc.RequiredDocker {
+		tc.Command.Flags[testutil.StartFlagDisableDocker] = "true"
+	}
+	cmd := testutil.GinkgoGetHorusecCmdWithFlags(testutil.CmdStart, tc.Command.Flags)
+	return gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
+}
+
+func NewTestCase() []TestCase {
+	return []TestCase{
+		{
+			Tool:           tools.HorusecEngine,
+			RequiredDocker: false,
+			Command: Command{
+				Flags: map[string]string{
+					testutil.StartFlagProjectPath: testutil.ExamplesPath,
+				},
+			},
+			Expected: Expected{
+				Language: languages.Generic,
+				ExitCode: 0,
+				OutputsContains: []string{
+					fmt.Sprintf(messages.MsgPrintFinishAnalysisWithStatus, analysis.Success),
+					messages.MsgDebugVulnHashToFix,
+					messages.MsgWarnAnalysisFoundVulns[16:],
+					"In this analysis, a total of 66 possible vulnerabilities were found and we classified them into:",
+					"Total of Vulnerability CRITICAL is: 17",
+					"Total of Vulnerability HIGH is: 22",
+					"Total of Vulnerability MEDIUM is: 24",
+					"Total of Vulnerability LOW is: 3",
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.CSharp),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Dart),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Java),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Javascript),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Kotlin),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Leaks),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Nginx),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Swift),
+					fmt.Sprintf("{HORUSEC_CLI} Running %s - %s", tools.HorusecEngine, languages.Yaml),
+					fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.GoSec),
+					fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.SecurityCodeScan),
+					fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.Brakeman),
+					fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.Safety),
+					fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.Bandit),
+					fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.NpmAudit),
+					fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.YarnAudit),
+					// TODO: This log show only if pass --enable-git-history, we can see if this tool was runned or not without this flag
+					//fmt.Sprintf("{HORUSEC_CLI} The tool was ignored for run in this analysis: %s", tools.GitLeaks),
```

Comment:
I don't understand this TODO and why this code is commented.

Reply:
in gitleaks not show this log `{HORUSEC_CLI} The tool was ignored for run in this analysis: GitLeaks` without the flag `--enable-git-history` enabled, is necessary add this log when user ignore this tool and when pass this flag with your value false

Reply:
same think for others TODO's